naptime/src/main/scala/org/coursera/naptime/errors.scala

```diff
@@ -46,6 +46,16 @@ case class NaptimeActionException(
       HttpEntity.Strict(ByteString.fromString(bodyString), Some(MimeTypes.JSON)))
   }
 
+  /**
+   * Set detail in NaptimeActionException which can be used for reasoning for downstream service.
+   * TODO: only exposing detail to internal users?
+   * @return NaptimeActionException with details set up as json formated detail.
+   */
+  def withExceptionDetail[T](detail: T)(implicit format: OFormat[T]): NaptimeActionException =
```

Comment:
Name the method `withExceptionDetails` and the parameter `details` to maintain consistency with the pull request title.